### PR TITLE
PDORNG - 1146 , Publish  Revenue data in Mobile Money Payments

### PR DESCRIPTION
**Pull request by KaveeshBaddage**
This com.apigate.bizao.revenue.publish.call.Template  is referring  MOBILE_MONEY_DB to get revenue_api_url FROM [global_settings table](https://github.com/WSO2Telco/bizao-mobile-money-API/pull/127/files#diff-9c27826e816538f938f87a32436dffc1).

### CODE REVIEW

File: bizao-revenue-publisher/src/main/synapse-config/templates/com.apigate.bizao.revenue.publish.call.Template.xml [hunk 1]
```diff
@@ -1,4 +1,84 @@
...
+        <!--Get revenue publish URL-->
+        <dblookup>
+            <connection>
+                <pool>
+                    <dsName>jdbc/MOBILE_MONEY_DB</dsName>
+                </pool>
+            </connection>
+            <statement>
+                <sql>
+                    <![CDATA[ SELECT revenue_api_url FROM global_settings WHERE id = 1 ]]></sql>
```
Comment (dsulochana) [inline bizao-revenue-publisher/src/main/synapse-config/templates/com.apigate.bizao.revenue.publish.call.Template.xml]: rename revenue_publish_api_url

File: bizao-revenue-publisher/src/main/synapse-config/templates/com.apigate.bizao.revenue.publish.call.Template.xml [hunk 2]
```diff
@@ -1,4 +1,84 @@
...
+                <result column="revenue_api_url" name="REVENUE_PUBLISH_URL"/>
+            </statement>
+        </dblookup>
+
```
Comment (dsulochana) [inline bizao-revenue-publisher/src/main/synapse-config/templates/com.apigate.bizao.revenue.publish.call.Template.xml:67]: Please add db changes for new column to global_settings

File: bizao-revenue-publisher/src/main/synapse-config/templates/com.apigate.bizao.revenue.publish.call.Template.xml [hunk 3]
```diff
@@ -1,4 +1,84 @@
...
+        <!-- Call revenue publish endpoint-->
+        <header name="To" expression="$ctx:REVENUE_PUBLISH_URL" scope="default"/>
```
Comment (dsulochana) [inline bizao-revenue-publisher/src/main/synapse-config/templates/com.apigate.bizao.revenue.publish.call.Template.xml]: pass Authorization header

**Review by dsulochana — CHANGES_REQUESTED** [review]

File: bizao-revenue-publisher/src/main/synapse-config/templates/com.apigate.bizao.revenue.publish.call.Template.xml (hunk 3 above)
Comment (KaveeshBaddage) [reply bizao-revenue-publisher/src/main/synapse-config/templates/com.apigate.bizao.revenue.publish.call.Template.xml]: Added Authorization header 

`<property name="Authorization" expression="fn:concat('Basic ', $ctx:REVENUE_PUBLISH_API_BEARER_TOKEN)" scope="transport" type="STRING"/>`

File: bizao-revenue-publisher/src/main/synapse-config/templates/com.apigate.bizao.revenue.publish.call.Template.xml (hunk 2 above)
Comment (KaveeshBaddage) [reply bizao-revenue-publisher/src/main/synapse-config/templates/com.apigate.bizao.revenue.publish.call.Template.xml:67]: Added DB changes to db_script7.sql file in bizao-mobile-money-API repository 

`ALTER TABLE `mobilemoneydb`.`global_settings`
ADD COLUMN `revenue_publish_api_url` VARCHAR(225) NOT NULL AFTER `payment_resource_location`,
ADD COLUMN `revenue_publish_api_bearer_token` VARCHAR(225) NOT NULL AFTER `revenue_publish_api_url`;`

File: bizao-revenue-publisher/src/main/synapse-config/templates/com.apigate.bizao.revenue.publish.call.Template.xml (hunk 1 above)
Comment (KaveeshBaddage) [reply bizao-revenue-publisher/src/main/synapse-config/templates/com.apigate.bizao.revenue.publish.call.Template.xml]: renamed to revenue_publish_api_url

**Comment by KaveeshBaddage** [comment]
Did requested changes - https://github.com/WSO2Telco/bizao-revenue-publisher/pull/1/commits/107d4096c96c42570406b3c7685b73b86888adfd

**Review by dsulochana — APPROVED** [review]

**Pull request merged**